pkg/operator/controller/crl/crl_configmap.go

```diff
@@ -0,0 +1,295 @@
+package crl
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"strings"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// authorityKeyIdentifier is a certificate's authority key identifier.
+type authorityKeyIdentifier struct {
+	KeyIdentifier []byte `asn1:"optional,tag:0"`
+}
+
+// authorityKeyIdentifierOID is the ASN.1 object identifier for the authority
+// key identifier extension.
+var authorityKeyIdentifierOID = asn1.ObjectIdentifier{2, 5, 29, 35}
+
+// ensureCRLConfigmap ensures the client CA certificate revocation list
+// configmap exists for a given ingresscontroller if the ingresscontroller
+// specifies a client CA certificate bundle in which any certificates specify
+// any CRL distribution points.  Returns a Boolean indicating whether the
+// configmap exists, the configmap if it does exist, and an error value.
+func (r *reconciler) ensureCRLConfigmap(ctx context.Context, ic *operatorv1.IngressController, namespace string, ownerRef metav1.OwnerReference, haveClientCA bool, clientCAConfigmap *corev1.ConfigMap) (bool, *corev1.ConfigMap, error) {
+	haveCM, current, err := r.currentCRLConfigMap(ctx, ic)
+	if err != nil {
+		return false, nil, err
+	}
+
+	var oldCRLs map[string]*pkix.CertificateList
+	if haveCM {
+		if data, ok := current.Data["crl.pem"]; ok {
+			if crls, err := buildCRLMap([]byte(data)); err != nil {
+				log.Error(err, "failed to parse current client CA configmap", "namespace", current.Namespace, "name", current.Name)
+			} else {
+				oldCRLs = crls
+			}
+		}
+
+	}
+
+	var clientCAData []byte
+	if haveClientCA {
+		clientCABundleFilename := "ca-bundle.pem"
+		if data, ok := clientCAConfigmap.Data[clientCABundleFilename]; !ok {
+			return haveCM, current, fmt.Errorf("client CA configmap %s/%s is missing %q", clientCAConfigmap.Namespace, clientCAConfigmap.Name, clientCABundleFilename)
```

Comment:
What is the ongoing runtime behaviour when this happens - do we just continue to reconcile?